public/docs/ts/latest/tutorial/toh-pt5.jade

```diff
@@ -421,120 +503,171 @@ code-example(format='').
+  Let's have the `!{_ActivatedRoute}` service and the `HeroService` injected
+  into the constructor with their values saved in private fields:
+
++makeExcerpt('app/hero-detail.component.ts (constructor)', 'ctor')
+
 :marked
   We tell the class that we want to implement the `OnInit` interface.
-+makeExample('toh-5/ts/app/hero-detail.component.ts', 'implement')(format=".")
-:marked
-  Inside the `ngOnInit` lifecycle hook, we use the `params` observable to
-  extract the `id` parameter value from the `ActivateRoute` service
-  and use the `HeroService` to fetch the hero with that `id`.
-+makeExample('toh-5/ts/app/hero-detail.component.ts', 'ng-oninit', 'app/hero-detail.component.ts (ngOnInit)')(format=".")
-:marked
-  Notice how we extract the `id` by calling the `forEach` method
-  which will deliver our array of route parameters.
-+makeExample('toh-5/ts/app/hero-detail.component.ts', 'get-id')(format=".")
+
++makeExcerpt('app/hero-detail.component.ts', 'implement', '')(format=".")
+
+block ngOnInit
+  :marked
+    Inside the `ngOnInit` lifecycle hook, we use the `params` observable to
+    extract the `id` parameter value from the `ActivateRoute` service
+    and use the `HeroService` to fetch the hero with that `id`.
+
++makeExcerpt('app/hero-detail.component.ts', 'ngOnInit')
+
+block extract-id
+  :marked
+    Notice how we extract the `id` by calling the `forEach` method
+    which will deliver our !{_array} of route parameters.
+
+- var _str2int = _docsFor == 'dart' ? '<code>int.parse</code> static method' : 'JavaScript (+) operator'
 :marked
   The hero `id` is a number. Route parameters are *always strings*.
-  So we convert the route parameter value to a number with the JavaScript (+) operator.
+  So we convert the route parameter value to a number with the !{_str2int}.
 
   ### Add *HeroService.getHero*
+
   The problem with this bit of code is that `HeroService` doesn't have a `getHero` method!
   We better fix that quickly before someone notices that we broke the app.
 
   Open `HeroService` and add a `getHero` method that filters the heroes list from `getHeroes` by `id`:
-+makeExample('toh-5/ts/app/hero.service.ts', 'get-hero', 'app/hero.service.ts (getHero)')(format=".")
+
++makeExcerpt('app/hero.service.ts', 'getHero')
+
 :marked
-  Return to the `HeroDetailComponent` to clean up loose ends.
+  Let's return to the `HeroDetailComponent` to clean up loose ends.
 
   ### Find our way back
 
   We can navigate *to* the `HeroDetailComponent` in several ways.
   How do we navigate somewhere else when we're done?
 
   The user could click one of the two links in the `AppComponent`. Or click the browser's back button.
-  We'll add a third option, a `goBack` method that navigates backward one step in the browser's history stack
+  We'll add a third option, a `goBack` method that navigates backward one step in the browser's history stack.
+
++makeExcerpt('app/hero-detail.component.ts', 'goBack')
```

Comment:
This is an excerpt but has no "..." to indicate that code is missing. (It's missing imports, the OnInit implementation, `@Component`, ...)